packages/api/amplify_api/lib/src/graphql/ws/web_socket_connection.dart

```diff
@@ -357,11 +371,14 @@ class WebSocketConnection implements Closeable {
 
   /// Clear variables used to track subscriptions and other helper variables
   /// related to web socket connection.
-  void _resetConnectionVariables([int closeStatus = _defaultCloseStatus]) {
+  Future<void> _resetConnectionVariables([
+    int closeStatus = _defaultCloseStatus,
+  ]) async {
     final reason =
         closeStatus == _defaultCloseStatus ? 'client closed' : 'unknown';
 
-    _channel?.sink.done.whenComplete(() {
+    await _channel?.sink.close(closeStatus, reason);
+    await _channel?.sink.done.whenComplete(() {
```

Comment:
question: why await and call whenComplete w callback? I would think you could just `await _channel?.sink.done;` on this line, then call the other stuff on next line with no callback bc you are awaiting done.